gravatar-ui/src/main/java/com/gravatar/ui/components/atomic/SocialMedia.kt

```diff
@@ -0,0 +1,134 @@
+    // Force the Gravatar icon
+    mediaList.add(SocialMedia(profile.profileUrl().url, LocalIcon.Gravatar.name, icon = LocalIcon.Gravatar))
+    // List and filter the other accounts from the profile, keep the same order coming from UserProfile.accounts list
+    profile.accounts?.forEach { account ->
+        try {
+            if (LocalIcon.valueOf(account.shortname) != null) {
+                // Add local icon if the shortname exists in our predefined list
+                mediaList.add(SocialMedia(URL(account.url), account.name, icon = LocalIcon.valueOf(account.shortname)))
+            } else {
+                // Add a "remote" icon (using the url coming from the endpoint response)
+                mediaList.add(SocialMedia(URL(account.url), account.name, iconUrl = URL(account.iconUrl)))
+            }
+        } catch (e: MalformedURLException) {
+            // Ignore invalid account or icon URLs that could be returned by the endpoint
+        }
+    }
+    return mediaList
+}
+
+@Composable
+fun SocialIcon(media: SocialMedia, modifier: Modifier = Modifier) {
+    val uriHandler = LocalUriHandler.current
+    IconButton(
+        onClick = {
+            uriHandler.openUri(media.url.toString())
+        },
+        modifier = modifier,
+    ) {
+        if (media.icon != null) {
+            Icon(
+                imageVector = ImageVector.vectorResource(media.icon.imageResource),
+                modifier = Modifier.fillMaxSize(),
+                contentDescription = media.name,
+            )
+        } else {
+            AsyncImage(
+                model = ImageRequest.Builder(LocalContext.current)
+                    .data(media.iconUrl.toString())
+                    .decoderFactory(SvgDecoder.Factory())
+                    .build(),
+                contentDescription = media.name,
+                modifier = Modifier.fillMaxSize(),
+                contentScale = androidx.compose.ui.layout.ContentScale.FillHeight,
+            )
+        }
+    }
+}
+
+@Composable
+fun SocialIconRow(socialMedia: List<SocialMedia>, modifier: Modifier = Modifier, maxIcons: Int = 4) {
+    Row(modifier = modifier) {
+        socialMedia.take(maxIcons).forEach { media ->
+            SocialIcon(media = media, modifier = Modifier.size(32.dp))
+        }
+    }
+}
+
+@Composable
+fun SocialIconRow(profile: UserProfile, modifier: Modifier = Modifier, maxIcons: Int = 4) {
+    SocialIconRow(mediaList(profile), modifier, maxIcons)
+}
+
+@Preview(showBackground = true)
+@Composable
+fun SocialIconRowPreview() {
+    val userProfile = UserProfile(
+        hash = "",
+        accounts = listOf(
+            Account(name = "Mastodon", url = "https://example.com", shortname = "mastodon"),
+            Account(name = "Tumblr", url = "https://example.com", shortname = "tumblr"),
+            // Invalid url, should be ignored:
+            Account(name = "TikTok", url = "example.com", shortname = "tiktok"),
```

Comment:
💡 We can create some unit test for `fun mediaList(profile: UserProfile)` to test that cases.

Reply:
Good point, but maybe we can keep this one for documention purpose